source/fundamentals/connections/tls.txt

```diff
@@ -0,0 +1,199 @@
+you can set the ``InsecureSkipVerify`` field to ``true``.
+
+Setting the ``InsecureSkipVerify`` field to ``true`` disables
+both certificate and hostname validation.
+   
+.. warning::
+
+   Specifying this option in a production environment makes
+   your application insecure and potentially
+   vulnerable to expired certificates and foreign processes posing
+   as valid client instances.
+
+For a full list of client options, see :ref:`golang-connection-options`.
+
+.. _golang-configure-tls-certificates:
+
+Configure Certificates
+----------------------
+
+To successfully initiate a TLS request, an application must prove its
+identity by referencing cryptographic certificates. To connect to
+MongoDB with TLS, your certificates must be stored as PEM
+files.
+
+.. important::
+
+   For production use, we recommend that your MongoDB deployment uses valid
+   certificates generated and signed by the same certificate authority.
+   For testing, it can use self-signed certificates.
+
+The following list describes the components needed to establish
+a connection with TLS:
+
+.. list-table::
+   :header-rows: 1
+   :widths: 30 70
+
+   * - TLS Component
+     - Description
+
+   * - Certificate Authority (CA)
+     - One or more certificate authorities to
+       trust when making a TLS connection.
+
+   * - Client Certificate
+     - A digital certificate and a key that allows the server to verify the identity
+       of your application to establish an encrypted network connection.
+
+   * - Certificate Key
+     - The client certificate private key file. This key is often
+       included within the certificate file itself.
+
+   * - Passphrase
+     - The password to decrypt the private client key if it is encrypted.
+
+.. tip::
+   
+   To learn more about the PEM format, see the Wikipedia entry on
+   :wikipedia:`Privacy-Enhanced Mail <w/index.php?title=Privacy-Enhanced_Mail&oldid=1184023212>`.
+
+.. _golang-client-tls-connect:
+
+Reference Certificates in a Client
+----------------------------------
+
+You must reference your certificates in your ``ClientOptions``
+object so that the server can validate them before the client connects.
+We recommend that you set the ``TLSConfig`` field of your
+``ClientOptions`` instance to a ``Config`` struct to configure your
+TLS connection. ``Config`` structs are native to Go and allow you to keep
+all your TLS options in a single reusable object.
+
+To create a ``Config`` instance, import the ``crypto/tls`` and
+``crypto/x509`` packages. Next, create a ``Config`` struct instance and
+set the relevant struct fields for your configuration.
+
+To learn more about the ``Config`` struct, see the `tls.Config API
+documentation <https://pkg.go.dev/crypto/tls#Config>`__.
+
+.. _golang-tls-config-full-example:
+
+Example
+~~~~~~~
+
+This example shows how to create a ``Config`` instance and
+a ``Client`` instance that includes TLS options by performing the
+following actions:
```

Comment:
S: could be cleaner to rewrite this paragraph to avoid starting every list entry with a participle

something like:

```suggestion
This example performins the
following actions to create a ``Config`` instance and
a ``Client`` instance that includes TLS options: 
```

Reply:
Should I shorten the code example to only include the body of main()? Or do you think showing the imports is helpful? Not usually shown in fundamentals code examples.

Reply:
i think it's helpful. i can imagine users saying the code sample doesn't work as written.